devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -172,7 +173,20 @@ public void execute() throws MojoExecutionException {
             final AppArtifact appArtifact = new AppArtifact(projectArtifact.getGroupId(), projectArtifact.getArtifactId(),
                     projectArtifact.getClassifier(), projectArtifact.getArtifactHandler().getExtension(),
                     projectArtifact.getVersion());
-            appArtifact.setPaths(PathsCollection.of(projectArtifact.getFile().toPath()));
+
+            File projectFile = projectArtifact.getFile();
```

Comment:
```suggestion
            File projectDir = projectArtifact.getFile();
```

Reply:
It's not necessarily a dir.

Reply:
If the project went through the `package` phase then it will be the JAR.